### PR TITLE
Remove misplaced call to genUnspillRegIfNeeded

### LINKED ISSUE #5047: RyuJIT x86: Float returns are not reloaded correctly
Sample code:
```C#
    [MethodImpl(MethodImplOptions.NoInlining)]
    public static float Get()
    {
        return 42;
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    public static float Test()
    {
        float x = Get();
        x -= Get();
        return x;
    }
```
RyuJIT x86 generates the following code:
```
       FF15C0451E01 call     [Program:Get():float]
       D95DFC       fstp     dword ptr [ebp-04H]
       F30F1045FC   movss    xmm0, dword ptr [ebp-04H]
       FF15C0451E01 call     [Program:Get():float]
       D95DFC       fstp     dword ptr [ebp-04H]
       F30F1045FC   movss    xmm0, dword ptr [ebp-04H]
       F30F5CC8     subss    xmm1, xmm0
       F30F114DFC   movss    dword ptr [ebp-04H], xmm1
       D945FC       fld      dword ptr [ebp-04H]
```
Note how `xmm1` is used without ever being loaded.

As far as I can tell this happens because `genCallInstruction` eagerly reloads and in doing so it uses the wrong register - the destination register of the call (which it seems to always be `xmm0`) instead of the destination register of the reload node that follows (which is `xmm1` in this case).

```
***** BB01, stmt 1 (top level)
     ( 34, 15) [000005] ------------             *  stmtExpr  void  (top level) (IL 0x000...0x00B)
N003 (  1,  4) [000006] ------------             |              /--*  const(h)  int    0x10945C0 ftn REG NA
N005 (  3,  6) [000007] ------------ control expr|           /--*  indir     int    REG NA
N007 ( 14,  5) [000001] --C-G------Z             |        /--*  call      float  Program.Get
     ( 14,  5) [000010] --C-G-------             |     /--*  reload    float  REG mm1
N009 (  1,  4) [000008] ------------             |     |     /--*  const(h)  int    0x10945C0 ftn REG NA
N011 (  3,  6) [000009] ------------ control expr|     |  /--*  indir     int    REG NA
N013 ( 14,  5) [000002] --C-G-------             |     +--*  call      float  Program.Get
N015 ( 33, 14) [000003] --C-G-------             |  /--*  -         float  REG mm1
N017 ( 34, 15) [000004] --C-G-------             \--*  return    float  REG NA
```

I suspect that `genCallInstruction` should not call `genUnspillRegIfNeeded` at line [5926](https://github.com/dotnet/coreclr/blob/master/src/jit/codegenxarch.cpp#L5917), that should happen when the user of the reload node consumes the register.

If `genUnspillRegIfNeeded` is removed then the following code is generated:
```
       FF15C0451C01 call     [Program:Get():float]
       D95DFC       fstp     dword ptr [ebp-04H]
       FF15C0451C01 call     [Program:Get():float]
       D95DF8       fstp     dword ptr [ebp-08H]
       F30F104DFC   movss    xmm1, dword ptr [ebp-04H]
       F30F1045F8   movss    xmm0, dword ptr [ebp-08H]
       F30F5CC8     subss    xmm1, xmm0
       F30F114DF8   movss    dword ptr [ebp-08H], xmm1
       D945F8       fld      dword ptr [ebp-08H]
```

### PR DESCRIPTION
**Pull request by mikedn**
As far as I can tell there's no reason for `genUnspillRegIfNeeded` to be called right after spilling the float return of a call, `genConsumeReg` will do this anyway at the appropriate place (and that includes a possible `GT_RELOAD` inserted by LSRA). It's also incorrect to unspill there as LSRA doesn't have a clue that the register will be in use at that point.

That said, I'm far from being familiar with register allocation so maybe I'm missing something. Though if there's something wrong with all this I suspect that it's probably caused by spilling, not unspilling.

Fixes #5047

### CODE REVIEW

**Comment by CarolEidt** [comment]
Good catch; looks good to me. I was the guilty party, and it apparently didn't occur to me that the reload should just be left to genConsumeReg. Thanks.

**Comment by CarolEidt** [comment]
I'm pretty sure the ARM build failures are unrelated, as they don't even use this file ;-)

**Comment by mikedn** [comment]
Do you think I should add a test for this? I actually found this due to a failing test - JIT\Methodical\fp\exgen\10w5d_cs_do\10w5d_cs_do.cmd - in #4986 but it's quite difficult to investigate failures in that test. Perhaps a targeted one would be useful, unless you're concerned about the ever increasing number of tests.

**Comment by CarolEidt** [comment]
I'm not concerned about the ever-increasing number of tests, especially when they are small, more focused ones. (Though that particular test has been particularly useful in finding bugs over the years!)

**Comment by mikedn** [comment]
Looks like those ARM builds passed after a rebase, probably just a glitch in CI.

**Comment by CarolEidt** [comment]
@dotnet/jit-contrib could someone else review this so that I can merge it?

File: tests/src/JIT/Regression/JitBlue/GitHub_5047/project.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{
```
Comment (russellhadley) [inline tests/src/JIT/Regression/JitBlue/GitHub_5047/project.json]: Can this test rely on one of the stock project.json?  There are other tests that don't need this.  @pgavlin I thought we were not using individual project.json files in the test tree.

Comment (pgavlin) [inline tests/src/JIT/Regression/JitBlue/GitHub_5047/project.json]: We do use individual project.json files for tests that have dependencies that are not included in the common projects.

Comment (pgavlin) [inline tests/src/JIT/Regression/JitBlue/GitHub_5047/project.json]: I do think that this can probably get away with the minimal project.json, however.

Comment (mikedn) [inline tests/src/JIT/Regression/JitBlue/GitHub_5047/project.json]: Thanks, I didn't realize that there's a common project.json. I'll update.

Comment (mikedn) [inline tests/src/JIT/Regression/JitBlue/GitHub_5047/project.json]: Shouldn't there be a common app.config as well? I see that many tests have an app.config file but it's pretty much the same.

Comment (pgavlin) [inline tests/src/JIT/Regression/JitBlue/GitHub_5047/project.json]: It wouldn't hurt. The reason that we have common project.json files is to speed up package restore--common app.configs don't have any similar compile or run time benefit that I'm aware of, so we did not do that work earlier.

**Comment by pgavlin** [comment]
LGTM.

**Pull request merged**